### PR TITLE
Optix bug reproduction - input & output denoiser DDS data

### PR DESCRIPTION
**Pull request by AnastaZIuk**
I will end it tomorrow

### CODE REVIEW

File: include/nbl/builtin/glsl/ext/LumaMeter/common.glsl [hunk 1]
```diff
@@ -52,6 +52,13 @@ struct nbl_glsl_ext_LumaMeter_Uniforms_t
         #define _NBL_GLSL_EXT_LUMA_METER_SHARED_SIZE_NEEDED_ _NBL_GLSL_WORKGROUP_ARITHMETIC_SHARED_SIZE_NEEDED_
     #endif
 
+	#if _NBL_GLSL_EXT_LUMA_METER_MAX_LUMA_DEFINED_<0xffffffff
+		#error "1"
+	#endif
+	#if _NBL_GLSL_EXT_LUMA_METER_MIN_LUMA_DEFINED_<0xffffffff
+		#error "1"
+	#endif 
+
```
Comment (devshgraphicsprogramming) [inline include/nbl/builtin/glsl/ext/LumaMeter/common.glsl:61]: remove this

Comment (AnastaZIuk) [reply include/nbl/builtin/glsl/ext/LumaMeter/common.glsl]: done

File: examples_tests/39.DenoiserTonemapper/main.cpp [hunk 2]
```diff
@@ -15,6 +15,14 @@
 
 #include "CommonPushConstants.h"
 
+/*
+	Uncomment bellow define to perform
+	DDS input/output Denoiser Tonemapper's
+	data saving.
+*/
+
+#define DDSInputOutputDenoiserSave
```
Comment (devshgraphicsprogramming) [inline examples_tests/39.DenoiserTonemapper/main.cpp:24]: I dont need that code anymore, just needed the DDSes

Comment (devshgraphicsprogramming) [reply examples_tests/39.DenoiserTonemapper/main.cpp:24]: TODO

Comment (AnastaZIuk) [reply examples_tests/39.DenoiserTonemapper/main.cpp]: may merge

File: examples_tests/39.DenoiserTonemapper/main.cpp [hunk 3]
```diff
@@ -65,7 +68,7 @@ int main(int argc, char* argv[])
 	params.Vsync = true;
 	params.Doublebuffer = true;
 	params.Stencilbuffer = false;
-	params.StreamingDownloadBufferSize = 256*1024*1024; // change in Vulkan fo
+	params.StreamingDownloadBufferSize = 2560*1024*1024;
```
Comment (devshgraphicsprogramming) [inline examples_tests/39.DenoiserTonemapper/main.cpp:71]: after generating the DDSes is there any need for such a ridiculously huge download buffer?

File: examples_tests/39.DenoiserTonemapper/main.cpp [hunk 4]
```diff
@@ -822,52 +814,93 @@ void main()
...
+					fakeScratchLink,
+					denoiser.scratchSize,
+					overlap,
+					tileWidth,
+					tileHeight
+				) != OPTIX_SUCCESS)
+				{
+					os::Printer::log(makeImageIDString(i) + "Could not invoke the denoiser sucessfully, skipping image!", ELL_ERROR);
+					continue;
+				}
```
Comment (devshgraphicsprogramming) [inline examples_tests/39.DenoiserTonemapper/main.cpp:903]: do we denoise with tiles and the bug or without tiles?

**Comment by devshgraphicsprogramming** [comment]
There's a conflict, please merge main repo `master` into your `master` first

File: examples_tests/39.DenoiserTonemapper/main.cpp (hunk 3 above)
Comment (AnastaZIuk) [reply examples_tests/39.DenoiserTonemapper/main.cpp:71]: no need actually

Comment (AnastaZIuk) [reply examples_tests/39.DenoiserTonemapper/main.cpp:71]: unfortunately we need more space though

File: examples_tests/39.DenoiserTonemapper/main.cpp (hunk 4 above)
Comment (AnastaZIuk) [reply examples_tests/39.DenoiserTonemapper/main.cpp:903]: tiles

**Comment by AnastaZIuk** [comment]
> There's a conflict, please merge main repo `master` into your `master` first

done, adjusted to new API (sorry for weird commit name, wrong quick paste)

File: examples_tests/39.DenoiserTonemapper/main.cpp (hunk 4 above)
Comment (devshgraphicsprogramming) [reply examples_tests/39.DenoiserTonemapper/main.cpp:903]: so we have the bug with the stripes?

File: examples_tests/39.DenoiserTonemapper/main.cpp (hunk 3 above)
Comment (devshgraphicsprogramming) [reply examples_tests/39.DenoiserTonemapper/main.cpp:71]: what for?

**Pull request merged**